legacy/builder/types/context.go

```diff
@@ -187,22 +187,6 @@ func (ctx *Context) ExtractBuildOptions() *properties.Map {
 	return opts
 }
 
-func (ctx *Context) InjectBuildOptions(opts *properties.Map) {
-	ctx.HardwareDirs = paths.NewPathList(strings.Split(opts.Get("hardwareFolders"), ",")...)
-	ctx.BuiltInToolsDirs = paths.NewPathList(strings.Split(opts.Get("builtInToolsFolders"), ",")...)
-	ctx.BuiltInLibrariesDirs = paths.NewPathList(strings.Split(opts.Get("builtInLibrariesFolders"), ",")...)
-	ctx.OtherLibrariesDirs = paths.NewPathList(strings.Split(opts.Get("otherLibrariesFolders"), ",")...)
-	ctx.SketchLocation = opts.GetPath("sketchLocation")
-	fqbn, err := cores.ParseFQBN(opts.Get("fqbn"))
-	if err != nil {
-		i18n.ErrorfWithLogger(ctx.GetLogger(), "Error in FQBN: %s", err)
-	}
-	ctx.FQBN = fqbn
-	ctx.ArduinoAPIVersion = opts.Get("runtime.ide.version")
-	ctx.CustomBuildProperties = strings.Split(opts.Get("customBuildProperties"), ",")
-	ctx.OptimizationFlags = opts.Get("compiler.optimization_flags")
-}
-
```

Comment:
Why did you remove this one? Is this function unused?

Reply:
Yep, completely unused. I was digging around trying to understand how to fix the issue and stumbled upon it, noticed it wasn't used and decided to delete since I was there.

Reply:
I checked, it's used in the `arduino-builder`, so we need it: https://github.com/arduino/arduino-builder/blob/642e2cd708dd024e658067d383387691f7e79abd/main.go#L244